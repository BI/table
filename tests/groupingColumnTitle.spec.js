/* eslint-disable no-undef */
import React from 'react';
import Table from '../';
import { render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';

describe('Table with grouping columns', () => {
  it('group columns', () => {
    /**
     * +---+---+---------------+-------+---+
     * |   |   |       C       |   J   |   |
     * |   |   +---+---------------+---+   |
     * |   |   |   |     E     |   |   |   |
     * | A | B |   +---+-------+   |   | M |
     * |   |   | D | F |   G   | K | L |   |
     * |   |   |   |   +---+---+   |   |   |
     * |   |   |   |   | H | I |   |   |   |
     * +---+---+---+---+---+---+---+---+---+
     */
    const columns = [
      { title: '表头A', className: 'title-a', dataIndex: 'a', key: 'a' },
      { title: '表头B', className: 'title-b', dataIndex: 'b', key: 'b' },
      { title: '表头C', className: 'title-c', children:
        [
          { title: '表头D', className: 'title-d', dataIndex: 'c', key: 'c' },
          { title: '表头E', className: 'title-e', children:
            [
              { title: '表头F', className: 'title-f', dataIndex: 'd', key: 'd' },
              { title: '表头G', className: 'title-g', children:
                [
                  { title: '表头H', className: 'title-h', dataIndex: 'e', key: 'e' },
                  { title: '表头I', className: 'title-i', dataIndex: 'f', key: 'f' },
                ],
              },
            ],
          },
        ],
      },
      { title: '表头J', className: 'title-j', children:
        [
          { title: '表头K', className: 'title-k', dataIndex: 'g', key: 'g' },
          { title: '表头L', className: 'title-l', dataIndex: 'h', key: 'h' },
        ],
      },
      { title: '表头M', className: 'title-m', dataIndex: 'i', key: 'i' },
    ];

    const data = [
      { key: '1', a: 'a1', b: 'b1', c: 'c1', d: 'd1', e: 'e1', f: 'f1', g: 'g1', h: 'h1', i: 'i1' },
      { key: '2', a: 'a2', b: 'b2', c: 'c2', d: 'd2', e: 'e2', f: 'f2', g: 'g2', h: 'h2', i: 'i2' },
      { key: '3', a: 'a3', b: 'b3', c: 'c3', d: 'd3', e: 'e3', f: 'f3', g: 'g3', h: 'h3', i: 'i3' },
    ];

    const wrapper = render(
      <Table columns={columns} data={data} />
    );

    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('work with fixed columns', () => {
    const columns = [
      { title: '表头A', className: 'title-a', dataIndex: 'a', key: 'a', fixed: 'left' },
      { title: '表头B', className: 'title-b', children:
        [
          { title: '表头C', className: 'title-c', dataIndex: 'b', key: 'b' },
          { title: '表头D', className: 'title-d', dataIndex: 'c', key: 'c' },
        ],
      },
      { title: '表头E', className: 'title-e', dataIndex: 'd', key: 'd', fixed: 'right' },
    ];

    const data = [
      { key: '1', a: 'a1', b: 'b1', c: 'c1', d: 'd1' },
      { key: '2', a: 'a2', b: 'b2', c: 'c2', d: 'd2' },
      { key: '3', a: 'a3', b: 'b3', c: 'c3', d: 'd3' },
    ];

    const wrapper = render(
      <Table columns={columns} data={data} />
    );

    expect(renderToJson(wrapper)).toMatchSnapshot();
  });
});
