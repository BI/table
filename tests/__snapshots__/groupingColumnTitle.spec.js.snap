exports[`Table with grouping columns group columns 1`] = `
<div
  class="rc-table rc-table-scroll-position-left">
  <div
    class="rc-table-content">
    <div
      class="">
      <span>
        <div
          class="rc-table-body"
          style="overflow-x:auto;">
          <table
            class="rc-table-fixed">
            <colgroup>
              <col />
              <col />
              <col />
              <col />
              <col />
              <col />
              <col />
              <col />
              <col />
            </colgroup>
            <thead
              class="rc-table-thead">
              <tr>
                <th
                  class="title-a"
                  rowspan="4">
                  表头A
                </th>
                <th
                  class="title-b"
                  rowspan="4">
                  表头B
                </th>
                <th
                  class="title-c"
                  colspan="4">
                  表头C
                </th>
                <th
                  class="title-j"
                  colspan="2">
                  表头J
                </th>
                <th
                  class="title-m"
                  rowspan="4">
                  表头M
                </th>
              </tr>
              <tr>
                <th
                  class="title-d"
                  rowspan="3">
                  表头D
                </th>
                <th
                  class="title-e"
                  colspan="3">
                  表头E
                </th>
                <th
                  class="title-k"
                  rowspan="3">
                  表头K
                </th>
                <th
                  class="title-l"
                  rowspan="3">
                  表头L
                </th>
              </tr>
              <tr>
                <th
                  class="title-f"
                  rowspan="2">
                  表头F
                </th>
                <th
                  class="title-g"
                  colspan="2">
                  表头G
                </th>
              </tr>
              <tr>
                <th
                  class="title-h">
                  表头H
                </th>
                <th
                  class="title-i">
                  表头I
                </th>
              </tr>
            </thead>
            <tbody
              class="rc-table-tbody">
              <tr
                class="rc-table-row  rc-table-row-level-0">
                <td
                  class="title-a">
                  <span
                    class="rc-table-row-indent indent-level-0"
                    style="padding-left:0px;" />
                  a1
                </td>
                <td
                  class="title-b">
                  b1
                </td>
                <td
                  class="title-d">
                  c1
                </td>
                <td
                  class="title-f">
                  d1
                </td>
                <td
                  class="title-h">
                  e1
                </td>
                <td
                  class="title-i">
                  f1
                </td>
                <td
                  class="title-k">
                  g1
                </td>
                <td
                  class="title-l">
                  h1
                </td>
                <td
                  class="title-m">
                  i1
                </td>
              </tr>
              <tr
                class="rc-table-row  rc-table-row-level-0">
                <td
                  class="title-a">
                  <span
                    class="rc-table-row-indent indent-level-0"
                    style="padding-left:0px;" />
                  a2
                </td>
                <td
                  class="title-b">
                  b2
                </td>
                <td
                  class="title-d">
                  c2
                </td>
                <td
                  class="title-f">
                  d2
                </td>
                <td
                  class="title-h">
                  e2
                </td>
                <td
                  class="title-i">
                  f2
                </td>
                <td
                  class="title-k">
                  g2
                </td>
                <td
                  class="title-l">
                  h2
                </td>
                <td
                  class="title-m">
                  i2
                </td>
              </tr>
              <tr
                class="rc-table-row  rc-table-row-level-0">
                <td
                  class="title-a">
                  <span
                    class="rc-table-row-indent indent-level-0"
                    style="padding-left:0px;" />
                  a3
                </td>
                <td
                  class="title-b">
                  b3
                </td>
                <td
                  class="title-d">
                  c3
                </td>
                <td
                  class="title-f">
                  d3
                </td>
                <td
                  class="title-h">
                  e3
                </td>
                <td
                  class="title-i">
                  f3
                </td>
                <td
                  class="title-k">
                  g3
                </td>
                <td
                  class="title-l">
                  h3
                </td>
                <td
                  class="title-m">
                  i3
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`Table with grouping columns work with fixed columns 1`] = `
<div
  class="rc-table rc-table-scroll-position-left">
  <div
    class="rc-table-content">
    <div
      class="rc-table-fixed-left">
      <span>
        <div
          class="rc-table-body-outer">
          <div
            class="rc-table-body-inner">
            <table
              class="rc-table-fixed">
              <colgroup>
                <col />
              </colgroup>
              <thead
                class="rc-table-thead">
                <tr>
                  <th
                    class="title-a"
                    rowspan="2">
                    表头A
                  </th>
                </tr>
              </thead>
              <tbody
                class="rc-table-tbody">
                <tr
                  class="rc-table-row  rc-table-row-level-0">
                  <td
                    class="title-a">
                    <span
                      class="rc-table-row-indent indent-level-0"
                      style="padding-left:0px;" />
                    a1
                  </td>
                </tr>
                <tr
                  class="rc-table-row  rc-table-row-level-0">
                  <td
                    class="title-a">
                    <span
                      class="rc-table-row-indent indent-level-0"
                      style="padding-left:0px;" />
                    a2
                  </td>
                </tr>
                <tr
                  class="rc-table-row  rc-table-row-level-0">
                  <td
                    class="title-a">
                    <span
                      class="rc-table-row-indent indent-level-0"
                      style="padding-left:0px;" />
                    a3
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </span>
    </div>
    <div
      class="rc-table-scroll">
      <span>
        <div
          class="rc-table-body"
          style="overflow-x:auto;">
          <table
            class="rc-table-fixed">
            <colgroup>
              <col />
              <col />
              <col />
              <col />
            </colgroup>
            <thead
              class="rc-table-thead">
              <tr>
                <th
                  class="title-a"
                  rowspan="2">
                  表头A
                </th>
                <th
                  class="title-b"
                  colspan="2">
                  表头B
                </th>
                <th
                  class="title-e"
                  rowspan="2">
                  表头E
                </th>
              </tr>
              <tr>
                <th
                  class="title-c">
                  表头C
                </th>
                <th
                  class="title-d">
                  表头D
                </th>
              </tr>
            </thead>
            <tbody
              class="rc-table-tbody">
              <tr
                class="rc-table-row  rc-table-row-level-0">
                <td
                  class="title-a">
                  <span
                    class="rc-table-row-indent indent-level-0"
                    style="padding-left:0px;" />
                  a1
                </td>
                <td
                  class="title-c">
                  b1
                </td>
                <td
                  class="title-d">
                  c1
                </td>
                <td
                  class="title-e">
                  d1
                </td>
              </tr>
              <tr
                class="rc-table-row  rc-table-row-level-0">
                <td
                  class="title-a">
                  <span
                    class="rc-table-row-indent indent-level-0"
                    style="padding-left:0px;" />
                  a2
                </td>
                <td
                  class="title-c">
                  b2
                </td>
                <td
                  class="title-d">
                  c2
                </td>
                <td
                  class="title-e">
                  d2
                </td>
              </tr>
              <tr
                class="rc-table-row  rc-table-row-level-0">
                <td
                  class="title-a">
                  <span
                    class="rc-table-row-indent indent-level-0"
                    style="padding-left:0px;" />
                  a3
                </td>
                <td
                  class="title-c">
                  b3
                </td>
                <td
                  class="title-d">
                  c3
                </td>
                <td
                  class="title-e">
                  d3
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </span>
    </div>
    <div
      class="rc-table-fixed-right">
      <span>
        <div
          class="rc-table-body-outer">
          <div
            class="rc-table-body-inner">
            <table
              class="rc-table-fixed">
              <colgroup>
                <col />
              </colgroup>
              <thead
                class="rc-table-thead">
                <tr>
                  <th
                    class="title-e"
                    rowspan="2">
                    表头E
                  </th>
                </tr>
              </thead>
              <tbody
                class="rc-table-tbody">
                <tr
                  class="rc-table-row  rc-table-row-level-0">
                  <td
                    class="title-e">
                    d1
                  </td>
                </tr>
                <tr
                  class="rc-table-row  rc-table-row-level-0">
                  <td
                    class="title-e">
                    d2
                  </td>
                </tr>
                <tr
                  class="rc-table-row  rc-table-row-level-0">
                  <td
                    class="title-e">
                    d3
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </span>
    </div>
  </div>
</div>
`;
